src/App.tsx

```diff
@@ -8,7 +8,69 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterBy } from './types/Filter';
+
+const filteredTodos = (todos: Todo[], query: string, filterBy: FilterBy) => {
+  let newTodos = [...todos];
+
+  if (query) {
+    const newQuery = query.toLowerCase().trim();
+
+    newTodos = newTodos.filter(todo => todo.title.toLowerCase().includes(newQuery));
+  }
+
+  switch (filterBy) {
+    case FilterBy.ACTIVE:
+      newTodos = newTodos.filter(todo => !todo.completed);
+      break;
+    case FilterBy.COMPLETED:
+      newTodos = newTodos.filter(todo => todo.completed);
+      break;
+    case FilterBy.ALL:
+    default:
+      break;
+  }
+
+  return newTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+
+  const [load, setLoad] = useState(true);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(true);
```
about boolean variables naming: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig